test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+    });
+
+    it("Is reverted if the caller is not whitelisted", async () => {
+      await expect(cardPaymentProcessor.connect(deployer).confirmPayments(
+        authorizationIds,
+        cashOutAccount.address
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_ACCOUNT_IS_NOT_WHITELISTED);
+    });
+
+    it("Is reverted if the payment authorization IDs array is empty", async () => {
+      await expect(cardPaymentProcessor.connect(admin).confirmPayments(
+        [],
+        cashOutAccount.address
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_INPUT_ARRAY_OF_AUTHORIZATION_IDS_IS_EMPTY);
+    });
+
+    it("Is reverted if one of the payment authorization IDs is zero", async () => {
+      await expect(cardPaymentProcessor.connect(admin).confirmPayments(
+        [authorizationIds[0], ZERO_BYTES16_STRING],
+        cashOutAccount.address
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_IS_ZERO);
+    });
+
+    it("Is reverted if one of the payments with provided authorization IDs does not exist", async () => {
+      await expect(cardPaymentProcessor.connect(admin).confirmPayments(
+        [
+          authorizationIds[0],
+          createBytesString(payments[payments.length - 1].authorizationId + 1, BYTES16_LENGTH)
+        ],
+        cashOutAccount.address
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_DOES_NOT_EXIST);
+    });
+
+    it("Is reverted if one of the payments is uncleared", async () => {
+      await proveTx(cardPaymentProcessor.connect(admin).unclearPayment(
+        authorizationIds[1]
+      ));
+      await expect(cardPaymentProcessor.connect(admin).confirmPayments(
+        authorizationIds,
+        cashOutAccount.address
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_IS_UNCLEARED);
+    });
+
+    it("Transfer tokens, emits the correct events, changes the state properly", async () => {
+      const expectedClearedBalance: number = 0;
+      await checkCardPaymentProcessorState(payments);
+      await expect(cardPaymentProcessor.connect(admin).confirmPayments(
+        authorizationIds,
+        cashOutAccount.address
+      )).to.changeTokenBalances(
+        brlcMock,
+        [cardPaymentProcessor, cashOutAccount, ...accountAddresses],
+        [-totalAmount, +totalAmount, ...accountAddresses.map(() => 0)]
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "ConfirmPayment"
+      ).withArgs(
+        authorizationIds[0],
+        payments[0].account.address,
+        payments[0].amount,
+        expectedClearedBalance,
+        payments[0].revocationCounter || 0
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "ConfirmPayment"
+      ).withArgs(
+        authorizationIds[1],
+        payments[1].account.address,
+        payments[1].amount,
+        expectedClearedBalance,
+        payments[1].revocationCounter || 0
+      );
+      payments.forEach(payment => payment.status = PaymentStatus.Confirmed)
```

Comment:
Codacy has a fix for the issue: [Replace `(payment·=>·payment.status·=·PaymentStatus.Confirmed)` with `((payment)·=>·(payment.status·=·PaymentStatus.Confirmed));`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
      payments.forEach((payment) => (payment.status = PaymentStatus.Confirmed));
```